aiven_db_migrate/migrate/pgmigrate.py

```diff
@@ -992,18 +1014,23 @@ def _migrate_roles(self) -> Dict[str, PGRoleTask]:
                     message=err.diag.message_primary,
                 )
             else:
-                if role.rolconfig:
-                    for conf in role.rolconfig:
-                        key, value = conf.split("=", 1)
-                        self.log.info("Setting config for role %r: %s = %s", role.rolname, key, value)
-                        self.target.c(f'ALTER ROLE {role.safe_rolname} SET "{key}" = %s', args=(value, ), return_rows=0)
-                roles[role.rolname] = PGRoleTask(
-                    rolname=rolname,
-                    rolpassword=role.rolpassword,
-                    status=PGRoleStatus.created,
-                    message="role created",
-                )
-
+                try:
+                    if role.rolconfig:
+                        for conf in role.rolconfig:
+                            key, value = conf.split("=", 1)
+                            self.log.info("Setting config for role %r: %s = %s", role.rolname, key, value)
+                            self.target.c(f'ALTER ROLE {role.safe_rolname} SET "{key}" = %s', args=(value, ), return_rows=0)
+                    roles[role.rolname] = PGRoleTask(
+                        rolname=rolname,
+                        rolpassword=role.rolpassword,
+                        status=PGRoleStatus.created,
+                        message="role created",
+                    )
+                # display warning when ProgrammingErrorERROR 42501: InsufficientPrivilege: permission denied to set parameter for a role
+                except psycopg2.errors.InsufficientPrivilege:
```

Comment:
If you choose to go for the "filtered roles" route, why do you still need this change? Isn't this overlapping?

Reply:
I hope to keep this change, with the filtered roles flag, user needs to know exactly what roles to filter before the migration and supply that to the cli argument, should there be any special setting with this role that is unable to be set with avnadmin's permission it would stop the migration as the exception is not caught.  I think failing to set a role should not stop the migration process completely.

Reply:
The problem is... it's quite a bold statement. We'd need to be sure that such role is not needed in the destination database. If we go on and then we discover after months that the migrated db doesn't work anymore, that's an issue. It's better to fail fast.

Again, this will impact everything: aiven_db_migrate is a **library** that could be used anywhere (and it's used in aiven-core as well, as a library), we're adding a very generic workaround for a very narrow use case.

This is an AWS-specific thing. Maybe an idea could be add a specific flag to the **command line**, something like ```ignore-aws-roles```, and you put AWS-specific roles there, and pass them to filtered roles.

Reply:
that sounds like a good idea, I could just add `rdstopmgr` to the filtered_roles if `--ignore-source-aws-roles` is specified, do you think we should include all the `rds*` roles when the flag specified? and a similar flag such as `--ignore-source-cloudsql-roles`? one concern is if aws and google changes the names of these roles, probably not something we want to maintain in the migration tool code...